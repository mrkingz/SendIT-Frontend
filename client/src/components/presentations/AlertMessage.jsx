import React, { Fragment } from "react";
import { PropTypes } from "prop-types";
import { connect } from "react-redux";
import ReactHtmlParser from "react-html-parser";
import classnames from "classnames";

export const AlertMessage = props => {
  return (
    <Fragment>
      {props.message ? (
        <div className="control-group">
          <div
            className={classnames({
              [`alert ${props.styles}`]: !!props.styles
            })}
          >
            {ReactHtmlParser(props.message)}
          </div>
        </div>
      ) : (
        ""
      )}
    </Fragment>
  );
};
AlertMessage.propTypes = {
  message: PropTypes.string,
  styles: PropTypes.string
};

export const mapStateToProps = ({ messageReducer }) => {
  return {
    message: messageReducer.message,
    styles: messageReducer.styles
  };
};

export default connect(
  mapStateToProps,
  null
)(AlertMessage);
