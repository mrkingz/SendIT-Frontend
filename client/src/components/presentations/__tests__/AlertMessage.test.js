import React from "react";
import { shallow } from "enzyme";
import { AlertMessage, mapStateToProps } from "../AlertMessage";

const props = {
  message: "Message",
  styles: "style"
};
const mockStore = {
  messageReducer: {
    message: "Message",
    styles: "style"
  }
};
const wrapper = shallow(<AlertMessage {...props} />);

describe("Alert Message", () => {
  it("should render Alert Message", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should return message", () => {
    const mapStateResult = mapStateToProps(mockStore);
    expect(mapStateResult.message).toEqual("Message");
  });

  it("should render when there is no message", () => {
    wrapper.setProps({
      message: null
    });
    expect(wrapper).toMatchSnapshot();
  });
});
