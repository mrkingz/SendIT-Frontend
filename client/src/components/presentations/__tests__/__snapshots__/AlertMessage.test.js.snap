// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Message should render Alert Message 1`] = `
<Fragment>
  <div
    className="control-group"
  >
    <div
      className="alert style"
    >
      Message
    </div>
  </div>
</Fragment>
`;

exports[`Alert Message should render when there is no message 1`] = `<Fragment />`;
